StandardCollector.py

```diff
@@ -0,0 +1,43 @@
+import logging
+from typing import Callable, Dict, List
+from dataclasses import dataclass
+from prometheus_client.core import Summary
+from prometheus_client.metrics_core import Metric
+import BaseCollector
+
+LOG = logging.getLogger('apic_exporter.exporter')
+
+
+@dataclass
+class CustomMetric:
+    name: str
+    query: str
+    process_data: Callable[[str, Dict], List[Metric]]
+
+
+class StandardCollector(BaseCollector.BaseCollector):
```

Comment:
I would go with different naming here, parallel to python Exceptions for example.
```
BaseCollector
Collector
XYZCollector
```

So renaming this to `Collector` would make sense to cover the generic use-case.